deeplearning4j-core/src/main/java/org/deeplearning4j/nn/layers/recurrent/GravesLSTM.java

```diff
@@ -187,9 +188,12 @@ public Gradient backwardGradient(INDArray derivative, Layer nextLayer, Gradient
 					.muli(Nd4j.getExecutioner().execAndReturn(Nd4j.getOpFactory().createTransform("tanh", zi).derivative()));
 			//Shape: [m,n^L]
 
-			INDArray prevLayerActivationSlice = activation;
+
+			// TODO this was added to master assuming activation was still passed in but change on this branch is epsilon - needs verification
+
+			INDArray prevLayerActivationSlice = epsilon;
 			if (!is2dInput) {
-				prevLayerActivationSlice = activation.slice(t, 2);
+				prevLayerActivationSlice = epsilon.slice(t, 2);
```

Comment:
@AlexDBlack I merged in master with the new Graves change but it uses activation which we are no longer pulling into this method as a parameter. Looked like you had switched it to epsilon before. Can you take a look at this change and see if it sounds correct. On master and this branch 2 Grave' tests fail regarding linear index on weights; thus, doesn't seem related.